### ISSUE TITLE
Add support for FST (fast signal trace) files

### ISSUE THREAD

**Issue by alanvgreen** [ThePerfectComputer/FastWaveBackend#7]
The FST format is often used when working with large traces. It is more compact and faster than VCD.

https://github.com/semify-eda/fstdumper is an FST reader/writer library.

**Comment by ThePerfectComputer** [comment]
Yes, support for FST is being considered. One thing I'd really like to do however is keep only Rust code in the codebase.

How hard would it be to convert fstdumper to Rust, if I were to use fstdumper as a reference for a Rust version?

**Comment by alanvgreen** [comment]
I'm sure you'll have a better idea than me about how much effort it is.

It would be less effort to wrap the C, of course :)

**Comment by oscargus** [comment]
FYI, there is an alternative library for FST (released after this was opened): https://docs.rs/fstapi/latest/fstapi/ 

(Although, naturally, it would be nice to have the same API etc independent of underlying file format.)
